src/prefect/cli/run.py

```diff
@@ -1,42 +1,688 @@
+        except FileNotFoundError:
+            raise TerminalError(
+                f"Parameter file does not exist: {os.path.abspath(param_file)!r}"
+            )
+        except ValueError as exc:
+            raise TerminalError(
+                f"Failed to parse JSON at {os.path.abspath(param_file)!r}: {exc}"
+            )
+
+    cli_params = load_json_key_values(params, "parameter")
+    conflicting_keys = set(cli_params.keys()).intersection(file_params.keys())
+    if conflicting_keys:
+        quiet_echo(
+            "The following parameters were specified by file and CLI, the CLI value "
+            f"will be used: {conflicting_keys}"
+        )
+    params_dict = {**file_params, **cli_params}
+
+    # Local flow run -------------------------------------------------------------------
+
+    if path or module:
+        # We can load a flow for local execution immediately if given a path or module,
+        # otherwise, we'll lookup the flow then pull from storage for a local run
+        with try_error_done("Retrieving local flow...", quiet_echo, traceback=True):
+            flow = get_flow_from_path_or_module(path=path, module=module, name=name)
+
+        # Set the desired log level
+        if no_logs:
+            log_level = 100  # CRITICAL is 50 so this should do it
+
+        run_info = ""
+        if params_dict:
+            run_info += f"└── Parameters: {params_dict}\n"
+        if context_dict:
+            run_info += f"└── Context: {context_dict}\n"
+
+        if run_info:
+            quiet_echo("Configured local flow run")
+            quiet_echo(run_info, nl=False)
+
+        quiet_echo("Running flow locally...")
+        from prefect.utilities.logging import temporary_logger_config
+
+        with temporary_logger_config(
+            level=log_level,
+            stream_fmt="└── %(asctime)s | %(levelname)-7s | %(message)s",
+            stream_datefmt="%H:%M:%S",
+        ):
+            with prefect.context(**context_dict):
+                try:
+                    result_state = flow.run(parameters=params_dict)
+                except Exception as exc:
+                    quiet_echo()
+                    log_exception(exc, indent=2)
+                    raise TerminalError("Flow run failed!")
+
+        if result_state.is_failed():
+            quiet_echo("Flow run failed!", fg="red")
+        else:
+            quiet_echo("Flow run succeeded!", fg="green")
+
+        return
+
+    # Backend flow run -----------------------------------------------------------------
+
+    client = Client()
+
+    # Validate the flow look up options we've been given and get the flow from the
+    # backend
+    with try_error_done("Looking up flow metadata...", quiet_echo):
+        flow_view = get_flow_view(
+            flow_or_group_id=flow_or_group_id,
+            project=project,
+            name=name,
+        )
+
+    if log_level:
+        run_config = flow_view.run_config
+        if not run_config.env:
```

Comment:
I shall not in this PR because `None` may have a different meaning than an empty dict and I want to separate touching a `RunConfig` base from here if I can. cc @jcrist on intent there

Reply:
We could normalize `None` to an empty dict in the object (you'd have to also handle this in the deserializer, since flows with null `env` already exist in prefect cloud's database). `None` and `{}` should be identical in meaning. I agree with Michael though that this doesn't need to be done here.